client/src/components/users/Users.unit.test.jsx

```diff
@@ -0,0 +1,85 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+//User-event dispatches the events like they would happen if a user interacted with the document.
+import userEvent from "@testing-library/user-event";
+import Users from "./Users";
+
+describe("should render text in <Users /> component", () => {
+  it("renders User Management text", () => {
+    render(<Users />);
+    const text = screen.getByText(/User Management/i);
+    expect(text).toBeInTheDocument();
+  });
+
+  it("renders list of users text", () => {
+    render(<Users />);
+    const text = screen.getByText(/List of Users/i);
+    expect(text).toBeInTheDocument();
+  });
+
+  it("renders add user text", () => {
+    render(<Users />);
+    const text = screen.getByText(/Add User/i);
+    expect(text).toBeInTheDocument();
+  });
+});
+
+describe("should render form  and form element", () => {
+  it("renders user form and submit button correctly", () => {
+    render(<Users />);
+    expect(screen.getByTestId("user-form")).toBeInTheDocument();
+
+    let buttonElement = screen.getByRole("button", {
+      name: /add/i,
+    });
+    expect(buttonElement).toBeInTheDocument();
+  });
+  it("render name input", () => {
+    render(<Users />);
+
+    const inputEl = screen.getByTestId("add-user-name");
```

Comment:
I would avoid using `getByTestId` and use `getByLabelText` instead. I think if you fix the `htmlFor` issue above, you should be able to do `screen.getByLabelText('Name')`